contrib/lua/src/lundump.c

```diff
@@ -234,7 +234,7 @@ static void fchecksize (LoadState *S, size_t size, const char *tname) {
 #define checksize(S,t)	fchecksize(S,sizeof(t),#t)
 
 static void checkHeader (LoadState *S) {
-  checkliteral(S, LUA_SIGNATURE + 1, "not a");  /* 1st char already checked */
+  checkliteral(S, (const char*)LUA_SIGNATURE + 1, "not a");  /* 1st char already checked */
```

Comment:
Not fan to modify 3rd party.
Those fixes should go in the origin code, and Premake should retrieve those external code in some ways (download/submodule/...)

Reply:
binmodules/luasocket is also a 3rd party.